src/components/Snackbar/useSnackbar.tsx

```diff
@@ -0,0 +1,80 @@
+import { useCallback } from "react";
+import { useSnackbarContext } from "src/components/Snackbar/SnackbarContext";
+import { SnackbarNoticeProps } from "src/components/Snackbar/SnackbarNotice";
+import { maybeCall } from "src/utils";
+
+export interface UseSnackbarHook {
+  triggerNotice: (props: TriggerNoticeProps) => { close: () => void };
+  closeNotice: (id: string) => void;
+}
+
+export function useSnackbar(): UseSnackbarHook {
+  const { setNotices } = useSnackbarContext();
+
+  const onClose = useCallback((noticeId: string) => {
+    setNotices((prev) => {
+      let returnValue = prev;
+
+      // check to see if the notice is still in our existing stack, if so then remove it. Otherwise it was manually closed, so return the existing value.
+      if (prev.some(({ id }) => id === noticeId)) {
+        returnValue = prev.filter(({ id }) => id !== noticeId);
+      }
+
+      // For good measure, reset the snackbarId when notices array is emptied.
+      if (returnValue.length === 0) {
+        snackbarId = 1;
+      }
+
+      return returnValue;
+    });
+  }, []);
+
+  const triggerNotice = useCallback(
+    (props) => {
+      // Sets `noticeId` to the current value of `snackbarId` and then increments.
+      const noticeId = props.id ?? `beamSnackbar:${snackbarId++}`;
+      let maybeTimeout: number | undefined;
+
+      if (!props.persistent) {
+        maybeTimeout = window.setTimeout(() => {
+          onClose(noticeId);
+          // Auto close in 10s
+        }, 10_000);
+      }
+
+      setNotices((prev) =>
+        prev.concat({
+          ...props,
+          id: noticeId,
+          onClose: () => {
+            // Because we reset the `snackbarId` if the notices array is empty, then we need to call `clearTimeout` to ensure we don't accidentally close the wrong notice
+            clearTimeout(maybeTimeout);
+            onClose(noticeId);
+            maybeCall(props.onClose);
+          },
+        }),
+      );
+
+      return {
+        close: () => {
+          // Because we reset the `snackbarId` if the notices array is empty, then we need to call `clearTimeout` to ensure we don't accidentally close the wrong notice
+          clearTimeout(maybeTimeout);
+          onClose(noticeId);
+        },
+      };
+    },
+    [onClose],
+  );
+
+  const closeNotice = useCallback((id: string) => onClose(id), [onClose]);
+
+  return { triggerNotice, closeNotice };
+}
+
+let snackbarId = 1;
+
+// Make `id` optional. Allows for implementations to systematically close an alert by a provided id.
+export interface TriggerNoticeProps extends Omit<SnackbarNoticeProps, "id" | "onClose"> {
+  id?: string;
+  onClose?: () => void;
```

Comment:
I would like to leave this for now. I think it'd be useful for the caller to know if the user dismissed the notice outside of the action. For instance, if this notice triggers a state change in the app and when the notice closes we want to revert that state change. This is potentially what will be happening for Schedule's V2's "moving tasks". The idea I heard was:
1. A task is chosen to move, which in turn triggers a snackbar notice. Why? They want this snackbar notice to all the user to keep context of which task they have chosen to move as they scroll through the schedule.
2. If the user chooses to close the notice via the notice's internal X button, then it would break them out of the "move task" flow. At that point we'd want to signal back to the app that end the "move in progress" state.

Granted, I am not sure this is the right place for this schedule task stuff to be signaling a move, however, I also don't think having this `onClose` hurts.